src/librustc_error_codes/lib.rs

```diff
@@ -3,7 +3,7 @@
 
 macro_rules! register_diagnostics {
     ($($ecode:ident: $message:expr,)* ; $($code:ident,)*) => (
-        pub static DIAGNOSTICS: &[(&str, &str)] = &[
+        pub static DIAGNOSTICS: &[(&str, Option<&str>)] = &[
             $( (stringify!($ecode), $message), )*
```

Comment:
It looks like at some point this macro was updated and we forgot about the $code argument. We should be able to avoid changing every line in error_codes.rs and instead have this be something like the following, I think.

```rust
$( (stringify!($ecode), Some($message)), )*
$( (stringify!($code), None), )*
```